### ISSUE TITLE
how to use the softmax loss with keras?

### ISSUE THREAD

**Issue by qinhaifangpku** [fchollet/keras#4155]
hi, all~

i am processing with 3D volume data segmentation.
i wonder if keras can use the softmax loss as the loss function in the model.compile function.

i read some git answer, it said use activation='softmax' in the last layer and loss = 'category_crossentropy' , but when i use it it when to an error said that the last layer can not use 'softmax' as activation because it just can gandle the data 2D or 3D, but my layer input is 5D(example,x,y,z,channel)

now i use the activation 'sigmoid' in the last layer,  and compile with loss = 'binary_crossentropy'
it can go.  but i wonder it is any other advice to implemente the softmax loss with my 5D output data?
is there any body have a good idea?

thank you in advance!

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [ 1] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [ 1] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [ 1] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by lizhiyuanUSTC** [comment]
i think softmax can only be used at full connected layers

**Comment by qinhaifangpku** [comment]
@lizhiyuanUSTC thank you! maybe i should have more lesson about deep learning! thank you!

**Comment by zxu7** [comment]
Actually you can implement softmax to the last layer of FCN. It was mentioned in this post:
https://github.com/fchollet/keras/issues/1169

**Comment by farizrahman4u** [comment]
You can reshape your 5D tensor to 2D/3D and apply softmax activation.
